stable/anchore-engine/templates/anchore_upgrade_role.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if .Values.anchoreGlobal.usePreupgradeHook -}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ template "anchore-engine.fullname" . }}-upgrade-role
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "anchore-engine.fullname" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- with .Values.anchoreGlobal.labels }}
+    {{ toYaml . | nindent 4 }}
+    {{- end }}
+  {{- if .Values.anchoreGlobal.annotations }}
+  annotations:
+    {{- with .anchoreGlobal.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- end }}
+rules:
```

Comment:
I'm assuming this is the minimum set of permissions needed for `kubectl scale`?

Reply:
correct. I also added `pods` so that a second initContainer can watch to make sure all the pods are actually down before completing.